starknet_py/hash/casm_class_hash.py

```diff
@@ -0,0 +1,62 @@
+import json
+from typing import List
+
+from starkware.cairo.common.poseidon_hash import poseidon_hash_many
+
+from starknet_py.cairo.felt import encode_shortstring
+from starknet_py.hash.utils import _starknet_keccak
+from starknet_py.net.client_models import CasmClass, CasmClassEntryPoint
+
+CASM_CLASS_VERSION = "COMPILED_CLASS_V1"
+
+
+def compute_casm_class_hash(casm_contract_class: CasmClass) -> int:
+    """
+    Calculate class hash of a CasmClass.
+    """
+    casm_class_version = encode_shortstring(CASM_CLASS_VERSION)
+
+    _entry_points = casm_contract_class.entry_points_by_type
+
+    external_entry_points_hash = poseidon_hash_many(
+        _entry_points_array(_entry_points.external)
+    )
+    l1_handler_entry_points_hash = poseidon_hash_many(
+        _entry_points_array(_entry_points.l1_handler)
+    )
+    constructor_entry_points_hash = poseidon_hash_many(
+        _entry_points_array(_entry_points.constructor)
+    )
+
+    _program_dict = dict(program=casm_contract_class.program)
+    hinted_casm_class_hash = _starknet_keccak(
+        data=json.dumps(_program_dict, sort_keys=True).encode()
+    )
+
+    _bytecode = [int(val, 0) for val in casm_contract_class.program["data"]]
+    bytecode_hash = poseidon_hash_many(_bytecode)
+
+    return poseidon_hash_many(
+        [
+            casm_class_version,
+            external_entry_points_hash,
+            l1_handler_entry_points_hash,
+            constructor_entry_points_hash,
+            hinted_casm_class_hash,
+            bytecode_hash,
+        ]
+    )
+
+
+def _entry_points_array(entry_points: List[CasmClassEntryPoint]) -> List[int]:
+    entry_points_array = []
+    for entry_point in entry_points:
+        assert entry_point.builtins is not None
```

Comment:
I'd prefer an explicit exception instead of assert here.

Reply:
I knew it!